lib/spider/scripts/in_context_learning.py

```diff
@@ -57,15 +65,35 @@ def main(cfg) -> None:
         }
         for data_point in dev_json
     ]
+    if cfg.prompts.template.count("{}") > 2:
+        train_json = json.load(open(DATASET_PATH / "train_spider.json"))
+        train_json = [
+            {
+                "db_id": data_point["db_id"],
+                "question": data_point["question"],
+                "query": data_point["query"],
+            }
+            for data_point in train_json
+        ]
+        train_questions = [data_point["question"] for data_point in train_json]
+        top_k = int((cfg.prompts.template.count("{}") - 2) / 2)
+        encoder_model = SentenceTransformer("all-mpnet-base-v2")
+        encoder_model.eval()
+        with torch.no_grad():
```

Comment:
Really not sure about this, but isn't `encoder_mode.eval()` redundant with `torch.no_grad()`? 😮 Maybe you can check? (in case of doubt, you can keep the redundancy, it doesn't really matter)